components/org.wso2.carbon.identity.application.authenticator.samlsso/src/main/java/org/wso2/carbon/identity/application/authenticator/samlsso/SAMLSSOAuthenticator.java

```diff
@@ -448,7 +448,8 @@ protected void initiateLogoutRequest(HttpServletRequest request,
                 throw new LogoutFailedException(e.getMessage(), e);
             }
         } else {
-            throw new UnsupportedOperationException();
+            // Fixed issue #5707 - Proper error message is needed when logout is not supported with SAML2 Federated Authenticator
+            throw new UnsupportedOperationException("Logout operation not supported with SAML2 Federated Authenticator");
```

Comment:
Changed the error message to "Logout operation not enabled in SAML2 Federated Authenticator"

Reply:
Fixed the wording